yarn-project/end-to-end/scripts/setup_canary.sh

```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+set -eu
+
+COMMIT_TAG=$1
+
+if [ -z "$COMMIT_TAG" ]; then
+  echo "No commit tag provided."
+  exit 0
+fi
+
+VERSION=$(npx semver $COMMIT_TAG)
+if [ -z "$VERSION" ]; then
+  echo "$COMMIT_TAG is not a semantic version."
+  exit 1
+fi
+
+echo "Removing all files & folders that aren't needed for canary tests"
+TARGET_DIR="./src"
+cd "$TARGET_DIR"
+
+# Loop through all files and folders in the directory
+for item in $(ls -A); do
+  if [[ "$item" != "index.ts" && "$item" != "canary" ]]; then
+    # Remove the item (either file or folder)
+    rm -rf "$item"
+  fi
+done
+cd ..
+
+echo "Updating external Aztec dependencies to version $VERSION"
+
+# Packages that are publically available in npm
+TARGET_PKGS=("@aztec/aztec.js" "@aztec/cli" "@aztec/l1-artifacts" "@aztec/noir-contracts")
```

Comment:
Thought of a better way to do this, grabbing the dependencies from the canary package rather than manually maintaining the list here, updating now